Kumi.Game/Online/API/APIConnection.cs

```diff
@@ -15,16 +17,45 @@ public partial class APIConnection : Component, IAPIConnectionProvider
     public Queue<APIRequest> RequestQueue { get; private set; } = new Queue<APIRequest>();
     public Bindable<APIAccount> LocalAccount { get; } = new Bindable<APIAccount>(new GuestAccount());
     public Bindable<APIState> State { get; } = new Bindable<APIState>(APIState.Offline);
-    public string SessionToken { get; } = null!;
+    public string SessionToken { get; set; }
 
     public APIConnection(ServerConfiguration configuration)
     {
         EndpointConfiguration = configuration;
-        serverConnector = new ServerConnector(this);
+        var serverConnector = new ServerConnector(this);
+        this.serverConnector = serverConnector;
+        
+        serverConnector.Closed += b =>
+        {
+            if (b)
+            {
+                // The client was intentionally closed.
+                return;
+            }
+            
+            // Attempt to reconnect.
+            Scheduler.AddDelayed(() =>
+            {
+                this.serverConnector.AuthorizationToken = string.Empty; // Tokens are invalid after successful connection.
+                var tokenReq = new WebsocketTokenRequest();
+                
+                tokenReq.Success += () =>
+                {
+                    serverConnector.AuthorizationToken = tokenReq.Response.GetToken();
+                    registerConnectorHandlers();
+                    
+                    // try and connect to the server.
+                    serverConnector.Start();
+                };
+                
+                Perform(tokenReq);
+            }, 5000L);
```

Comment:
Why are we performing this 5 seconds later?

Reply:
Convenience, really. It only occurs whenever the disconnect was induced by the server closing; which allows it any possible allotted time to restart when necessary.

Reply:
If I'm reading this code correctly, it will retry only once and never again, until the server connector is closed again (which I don't think will be called if it failed to reconnect the first time around). Despite that, I don't see a reason why the delay should be so high, at most it should probably be `500ms`.

Reply:
I believe `1s` is a good compromise then, because the pain of the server handling 10+ reconnects at the same time from different clients is a valid concern.